x-pack/plugins/security_solution/scripts/endpoint/agent_downloader_cli/agent_downloader.ts

```diff
@@ -11,21 +11,63 @@ import type { ToolingLog } from '@kbn/tooling-log';
 import { getAgentDownloadUrl, getAgentFileName } from '../common/fleet_services';
 import { downloadAndStoreAgent } from '../common/agent_downloads_service';
 
+// Decrement the patch version by 1
+const decrementPatchVersion = (version: string): string => {
+  const [major, minor, patch] = version.split('.').map(Number);
+  return `${major}.${minor}.${patch - 1}`;
+};
+
+// Generate a list of versions to attempt downloading, including a fallback to the previous patch (GA)
+const getVersionsToDownload = (version: string): string[] => {
+  const patch = parseInt(version.split('.')[2], 10);
+
+  // If patch version is 0, return only the current version.
+  if (patch === 0) {
+    return [version];
+  }
+
+  return [version, decrementPatchVersion(version)];
+};
+// Download and store the Elastic Agent for the specified version(s)
 const downloadAndStoreElasticAgent = async (
   version: string,
   closestMatch: boolean,
   log: ToolingLog
-) => {
-  const downloadUrlResponse = await getAgentDownloadUrl(version, closestMatch, log);
-  const fileNameNoExtension = getAgentFileName(version);
-  const agentFile = `${fileNameNoExtension}.tar.gz`;
-  await downloadAndStoreAgent(downloadUrlResponse.url, agentFile);
+): Promise<void> => {
+  const versionsToDownload = getVersionsToDownload(version);
+
+  for (const v of versionsToDownload) {
+    try {
+      const { url } = await getAgentDownloadUrl(v, closestMatch, log);
+      const fileName = `${getAgentFileName(v)}.tar.gz`;
+
+      await downloadAndStoreAgent(url, fileName);
+      log.info(`Successfully downloaded and stored version ${v}`);
+      return; // Exit once successful
```

Comment:
So you download just the first one? and then just exit? you are inside of a loop here, so I'm confused by this `return` here

Reply:
Great question! The reason we don’t download multiple versions is in most cases, only one version of the agent is needed to complete the task. The `getVersionsToDownload` function generates a list of potential versions (the exact version requested and a fallback), but the intention is to find the first valid version that can be successfully downloaded. Once that happens, downloading multiple versions would be redundant.
Was my thinking correct? Are you suggesting that we should download both versions?

Reply:
Ok. So you are still only looking ot download 1 version, even though you get a list of versions for download. 

Maybe add this information to the above `for...of` loop so that that is clear to others that might look at this code in the future.